### PR TITLE
Filter out Nanopore reads from Bracken

### LINKED ISSUE #167: Should we disable Bracken for long read data?
### Description of feature

Bracken expects a specific read length as input for which a database has to be preconstructed. Long reads (especially ONT) are typically quite variable in size, making it hard to decide on such a read length. Additionally, I would assume that the longer reads  are very specific to one particular reference genome such that maybe post-processing by Bracken is not needed at all.

Maybe we should disable Bracken for long reads?

### PR DESCRIPTION
**Pull request by jfy133**
Closes #167 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/profiling.nf [hunk 1]
```diff
@@ -145,7 +145,15 @@ workflow PROFILING {
     if ( params.run_kraken2 && params.run_bracken ) {
         // Remove files from 'pure' kraken2 runs, so only those aligned against Bracken & kraken2 database are used.
         def ch_kraken2_output = KRAKEN2_KRAKEN2.out.report
-            .filter { meta, report -> meta['tool'] == 'bracken' }
+            .filter {
+                meta, report ->
+                    if ( meta['instrument_platform'] == 'OXFORD_NANOPORE' ) log.warn "[nf-core/taxprofiler] Bracken has not been evaluated for Nanopore data. Skipping Bracken for sample ${meta.id}."
```
Comment (Midnighter) [inline subworkflows/local/profiling.nf:150]: It seems to me that you should first check for tool == bracken before issuing the warning.

File: subworkflows/local/profiling.nf [hunk 2]
```diff
@@ -145,7 +145,15 @@ workflow PROFILING {
...
+                    meta['tool'] == 'bracken' && meta['instrument_platform'] != 'OXFORD_NANOPORE'
```
Comment (Midnighter) [inline subworkflows/local/profiling.nf:151]: I think the same would apply to PacBio (do we allow that already? 🤔 )

**Review by Midnighter — CHANGES_REQUESTED** [review]
I think the condition for the warning is incorrect but I'm not 100% sure.

File: subworkflows/local/profiling.nf (hunk 2 above)
Comment (jfy133) [reply subworkflows/local/profiling.nf:151]: I don't think we support that anywhere atm, it's in the lits, but we don't check for that (I know even less about Pac Bio than nanopore tbh)

Comment (jfy133) [inline subworkflows/local/profiling.nf:151]:
```suggestion
                    meta['tool'] == 'bracken' && meta['instrument_platform'] != 'OXFORD_NANOPORE'
                    if ( meta['instrument_platform'] == 'OXFORD_NANOPORE' ) log.warn "[nf-core/taxprofiler] Bracken has not been evaluated for Nanopore data. Skipping Bracken for sample ${meta.id}."
```

File: subworkflows/local/profiling.nf (hunk 1 above)
Comment (Midnighter) [reply subworkflows/local/profiling.nf]: Did you test the new version? I always thought that the last value from a closure was returned.

File: subworkflows/local/profiling.nf [hunk 3]
```diff
@@ -145,7 +145,15 @@ workflow PROFILING {
...
         // Remove files from 'pure' kraken2 runs, so only those aligned against Bracken & kraken2 database are used.
         def ch_kraken2_output = KRAKEN2_KRAKEN2.out.report
-            .filter { meta, report -> meta['tool'] == 'bracken' }
+            .filter {
+                meta, report ->
+                    meta['tool'] == 'bracken' && meta['instrument_platform'] != 'OXFORD_NANOPORE'
+                    if ( meta['instrument_platform'] == 'OXFORD_NANOPORE' ) log.warn "[nf-core/taxprofiler] Bracken has not been evaluated for Nanopore data. Skipping Bracken for sample ${meta.id}."
+
+
+            }
```
Comment (Midnighter) [inline subworkflows/local/profiling.nf:154]: I had missed that you applied this filter to the data that will serve as the input for Bracken only. So in that sense the tool check is actually not needed. I do think the value to be returned from a closure needs to be the last statement, though. Have you tested this?

**Comment by jfy133** [comment]
Oops, I misunderstood your original comment about checking for Bracken first, gone back to the original (tested and working!) order (and tested again ;) ).

It is working as expected now. All I've done is extended what already exists. We still need to keep the tool check though to ensure we don't pick up the Kraken2 reports (as we went through when you originally added this, IIRC).

**Review by Midnighter — APPROVED** [review]
Would be cool to have nf-test for this but oh well. Thanks for the fix and sorry for the confusion.

**Comment by jfy133** [comment]
> Oops, I misunderstood your original comment about checking for Bracken first, gone back to the original (tested and working!) order (and tested again ;) ).
> 
> It is working as expected now. All I've done is extended what already exists. We still need to keep the tool check though to ensure we don't pick up the Kraken2 reports (as we went through when you originally added this, IIRC).

No worries, we'll get to it eventually ;)

**Pull request merged**